main.go

```diff
@@ -86,30 +86,50 @@ func assetDelivery(w http.ResponseWriter, r *http.Request) {
 	}
 
 	w.Header().Set("Content-Type", mime.TypeByExtension(ext))
+	w.Header().Set("X-Content-Type-Options", "nosniff")
 	w.Write(assetData)
 }
 
-func handleIndex(w http.ResponseWriter, r *http.Request) {
-	indexTpl, err := assets.ReadFile("frontend/index.html")
-	if err != nil {
-		http.Error(w, "404 not found", http.StatusNotFound)
-		return
-	}
+var (
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
+	cspHeader = strings.Join([]string{
+		"default-src 'none'",
+		"connect-src 'self'",
+		"font-src 'self'",
+		"img-src 'self'",
+		"script-src 'self' 'unsafe-inline'",
+		"style-src 'self' 'unsafe-inline'",
+	}, ";")
+
+	indexTpl *template.Template
+)
 
-	tpl, err := template.New("index.html").Funcs(tplFuncs).Parse(string(indexTpl))
+func init() {
+	source, err := assets.ReadFile("frontend/index.html")
 	if err != nil {
-		http.Error(w, errors.Wrap(err, "parsing template").Error(), http.StatusInternalServerError)
-		return
+		log.WithError(err).Fatal("frontend folder should contain index.html Go template")
 	}
+	indexTpl = template.Must(template.New("index.html").Funcs(tplFuncs).Parse(string(source)))
+}
 
-	if err = tpl.Execute(w, struct {
-		Vars map[string]string
+func handleIndex(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	w.Header().Set("Referrer-Policy", "no-referrer")
+	w.Header().Set("X-Frame-Options", "DENY")
+	w.Header().Set("X-Xss-Protection", "1; mode=block")
+	w.Header().Set("Content-Security-Policy", cspHeader)
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+
+	// Disable some features that may be abused on an app like this.
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy
+	w.Header().Set("Feature-Policy", "display-capture 'none'; web-share 'none'")
```

Comment:
Hmm this header is experimental, has been renamed and also `web-share` is not supported:
> Error with Feature-Policy header: Unrecognized feature: 'web-share'.

![image](https://user-images.githubusercontent.com/20142/134811661-267e042f-b3c7-493d-890b-1e66ebb4eadf.png)

Maybe we should remove this until it's no longer an "Editor’s Draft"…

Reply:
You're right. I was actually confused with the Permissions-Policy header which seems to be an already deprecated version of the same idea.